cirrus/server/tests/test_sdk.py

```diff
@@ -1,16 +1,78 @@
 import unittest
+from unittest.mock import patch
+
+from cirrus_sdk import NimbusError
 
 from cirrus.sdk import SDK
 
 
 class SDKTestCase(unittest.TestCase):
     def test_compute_enrollments(self):
         sdk = SDK()
-        recipes = [
-            {"id": 1, "name": "recipe1"},
-            {"id": 2, "name": "recipe2"},
-        ]
-        targeting_context = {"client_id": "testid"}
-
-        result = sdk.compute_enrollments(recipes, targeting_context)
-        self.assertEqual(result, [])
+        targeting_context = {"clientId": "test", "requestContext": {}}
+
+        result = sdk.compute_enrollments(targeting_context)
+        self.assertEqual(
+            result, {"enrolledFeatureConfigMap": {}, "enrollments": [], "events": []}
+        )
+
+    def test_failure_case_no_client_id_key(self):
+        sdk = SDK()
+        targeting_context = {"requestContext": {}}
+
+        try:
+            result = sdk.compute_enrollments(targeting_context)
+        except NimbusError:
+            self.fail("NimbusError was raised when it should not have been")
+        self.assertEqual(result, {})
+
+    def test_failure_case_no_client_id_value(self):
+        sdk = SDK()
+        targeting_context = {"clientId": None, "requestContext": {}}
+
+        try:
+            result = sdk.compute_enrollments(targeting_context)
+        except NimbusError:
+            self.fail("NimbusError was raised when it should not have been")
+        self.assertEqual(result, {})
+
+    def test_failure_case_no_request_context(self):
+        sdk = SDK()
+        targeting_context = {"clientId": "test"}
+
+        try:
+            result = sdk.compute_enrollments(targeting_context)
+        except NimbusError:
+            self.fail("NimbusError was raised when it should not have been")
+        self.assertEqual(result, {})
+
+    def test_set_experiments_success(self):
+        sdk = SDK()
+        recipes = '{"data": []}'
+
+        with patch.object(sdk.client, "set_experiments") as mock_set_experiments:
+            sdk.set_experiments(recipes)
+
+        mock_set_experiments.assert_called_once_with(recipes)
+
+    def test_set_experiments_failure(self):
+        sdk = SDK()
+        recipes = '{"data": []}'
+
+        with patch.object(sdk.client, "set_experiments") as mock_set_experiments:
```

Comment:
Why do we need to mock this out?

Reply:
Yeah like usually you need to mock out network traffic, but this should be entirely local so we should be able to get away with actually making this call?

Reply:
was writing test cases in the flow, changed :D